dvc/command/repro.py

```diff
@@ -42,14 +42,15 @@ def _repro_kwargs(self):
             "force_downstream": self.args.force_downstream,
             "pull": self.args.pull,
             "glob": self.args.glob,
+            "glob_stages": self.args.glob_stages,
         }
 
 
 def add_arguments(repro_parser):
     repro_parser.add_argument(
         "targets",
         nargs="*",
-        help="Stages to reproduce. 'dvc.yaml' by default.",
+        help="Stages or output files to reproduce. 'dvc.yaml' by default.",
```

Comment:
could be a dir.
```suggestion
        help="Stages or outputs to reproduce. 'dvc.yaml' by default.",
```

Reply:
Actually, can we avoid the term "output"? Per https://github.com/iterative/dvc.org/pull/2357#pullrequestreview-629433013 cc @shcheklein 

Maybe:
```suggestion
        help="Stages or data files or directories to reproduce. 'dvc.yaml' by default.",
```

Reply:
TBH I'm not sure it's undesirable to use the term "output" here, since it's part of the dvc.yaml schema (and probably in many UI strings of `dvc`), but let's try.

- [ ] Note for self: review other existing places where "output" can be avoided.

Reply:
From the user perspective it seems misleading to me to use "data files" term here. Current code doesn't let you to reproduce anything by dependencies names, which I would assume is possible after reading "data files" in help for targets. We could add more descriptive term like "... files defined as outputs from stages ..." there. What do you think?

Reply:
my 2cs - I think in this specific case `output files` are fine (note! output files != output)

Reply:
I mean ... may be even `outputs` are fine @jorgeorpinel :) If we indeed mean them in end-user meaning (a short cut to the results or output files or more precise form that @Vonski mentioned). What I'd like to avoid I guess, is when we use `outputs` for a command that takes a path to _any_ DVC-tracked file, model, directory. E.g. `dvc add`-ed files. They are called "outputs" internally (and even in the .dvc file), but from the readers/end user perspective those are datasets, data, or files, _input_ data (when there is a pipeline).

In this specific case (my understanding) that what we add is an ability to take any "pipeline produced output" and pass it into `repro`. In this case it's more or less clear I think - that yes, there are pipelines - they have inputs and outputs, and you can pass one of those.

Sorry, for a long description, was trying to unload my thought process :)

Reply:
Got it. So "outputs" should be OK here, but as an alternative we can go with "stage outputs", though in this case we would need something like the following (a little repetitive but clear):
```suggestion
        help="Stage names or stage outputs to reproduce. 'dvc.yaml' by default.",
```

Reply:
> 'dvc.yaml' by default.

BTW, that's neither a stage name nor a stage output. Should it be "Everything in /dvc.yaml by default" ?